amlb/datasets/openml.py

```diff
@@ -140,7 +146,7 @@ def attributes(self):
             log.debug("Loading attributes from dataset %s.", self._oml_dataset.data_file)
             with open(self._oml_dataset.data_file) as f:
                 ds = arff.load(f)
-                self._attributes = ds['attributes']
+                self._attributes = [a for a in ds['attributes'] if a[0] not in self._oml_dataset.ignore_attribute]
```

Comment:
What is the reason for loading the arff file instead of using the `OpenMLDataset.features`?
The `OpenMLDataset.ignore_attribute` can be `None`, I believe this will cause issues with the `in` keyword.

Reply:
legacy, it's a fallback mechanism for older version of openml-python (see comment line 113).
We can clean this up in a future PR.

Reply:
> The OpenMLDataset.ignore_attribute can be None, I believe this will cause issues with the in keyword.

good catch: fails on the test datasets indeed... fixing